### PR TITLE
EASEL-Creator-Hub-Screen-Adjustments-issue-#1280

### PR DESCRIPTION
**Pull request by Kamran-Khan-Gene**
**Made changes suggested by Tanyaml.**

1- Removed Publish button from published NFT in GridView.
2- Added NTF File Types Icons.
3- Added test cases to test NFT File Type Icons.

## Description

Closes: #XXXX

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by Kamran-Khan-Gene** [comment]
@kjawadDeveloper2 Kindly review my PR.

File: easel/lib/screens/creator_hub/widgets/nfts_grid_view.dart [hunk 1]
```diff
@@ -81,7 +81,24 @@ class NftGridViewItem extends StatelessWidget {
               child: Padding(
                 padding: EdgeInsets.only(left: 8.w, top: 10.0.h),
                 child: SvgPicture.asset(
-                  SVGUtils.kFileTypeImageIcon,
+                  nft.assetType == kImageText
```
Comment (kjawadDeveloper2) [inline easel/lib/screens/creator_hub/widgets/nfts_grid_view.dart:84]: Make a method instead of this ladder.

Comment (Kamran-Khan-Gene) [reply easel/lib/screens/creator_hub/widgets/nfts_grid_view.dart]: Updated..

File: easel/lib/screens/creator_hub/widgets/nfts_grid_view.dart [hunk 2]
```diff
@@ -81,7 +81,24 @@ class NftGridViewItem extends StatelessWidget {
...
+                      ? SVGUtils.kFileTypeImageIcon
+                      : nft.assetType == kVideoText
+                          ? SVGUtils.kSvgNftFormatVideo
+                          : nft.assetType == k3dText
+                              ? SVGUtils.kSvgNftFormat3d
+                              : nft.assetType == kPdfText
+                                  ? SVGUtils.kSvgNftFormatPDF
+                                  : SVGUtils.kSvgNftFormatAudio,
+                  key: Key(nft.assetType == kImageText
```
[the remark this answers is not in the document]
Comment (Kamran-Khan-Gene) [reply easel/lib/screens/creator_hub/widgets/nfts_grid_view.dart]: Done.

File: easel/lib/screens/creator_hub/widgets/nfts_grid_view.dart [hunk 3]
```diff
@@ -114,11 +131,15 @@ class NftGridViewItem extends StatelessWidget {
                   const Spacer(),
                   InkWell(
                     onTap: () {
-                      final DraftsBottomSheet draftsBottomSheet = DraftsBottomSheet(
-                        buildContext: context,
-                        nft: nft,
-                      );
-                      draftsBottomSheet.show();
+                      if (context.read<CreatorHubViewModel>().selectedCollectionType == CollectionType.draft) {
```
[the remark this answers is not in the document]
Comment (Kamran-Khan-Gene) [reply easel/lib/screens/creator_hub/widgets/nfts_grid_view.dart:119]: Done..

**Comment by Kamran-Khan-Gene** [comment]
@kjawadDeveloper2 kindly review my PR.

**Comment by Kamran-Khan-Gene** [comment]
@kjawadDeveloper2 kindly review my PR

**Review by kjawadDeveloper2 — APPROVED** [review]

File: easel/lib/screens/creator_hub/widgets/draft_list_tile.dart [hunk 4]
```diff
@@ -243,8 +243,8 @@ class _DraftListTileState extends State<DraftListTile> {
                   ),
                 ),
               )
-            : const IgnorePointer(
-                child: SizedBox(),
+            : IgnorePointer(
+                child: SizedBox(height: 85.h,),
```
Comment (ahmadrns) [inline easel/lib/screens/creator_hub/widgets/draft_list_tile.dart:247]: code formatting not done

File: easel/lib/screens/creator_hub/widgets/nfts_list_tile.dart [hunk 5]
```diff
@@ -218,8 +218,8 @@ class NFTsListTile extends StatelessWidget {
                   ),
                 ),
               )
-            : const IgnorePointer(
-                child: SizedBox(),
+            : IgnorePointer(
+                child: SizedBox(height: 85.h,),
```
Comment (ahmadrns) [inline easel/lib/screens/creator_hub/widgets/nfts_list_tile.dart:222]: code formatting not done

File: easel/test/mocks/mock_constants.dart [hunk 6]
```diff
@@ -44,5 +44,88 @@ NFT MOCK_PRICED_NFT = NFT(
...
+  recipeID: "Easel_Recipe_auto_recipe_2022_08_31_154526_206",
+  duration: "0:0",
+  cookbookID: "Easel_CookBook_auto_cookbook_2022_08_31_152836_312",
+  appType: "easel",
+  creator: "Ahmad",
+  fileSize: "90.12KB",
+  itemID: "DtnxAS8L4pf",
+  owner: "abd",
+  ibcCoins: IBCCoins.upylon.name,
+  assetType: "3D",
```
Comment (ahmadrns) [inline easel/test/mocks/mock_constants.dart:126]: Kindly assign assetType from existing enum

File: easel/test/widget/file_type_icon_test.dart [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+import 'package:provider/provider.dart';
+
+import '../extensions/size_extension.dart';
+import '../mock/mock_repository.dart';
+import '../mocks/mock_constants.dart';
+
+void main() {
+  TestWidgetsFlutterBinding.ensureInitialized();
+  final repository = MockRepositoryImp();
+  GetIt.I.registerLazySingleton(() => CreatorHubViewModel(repository));
```
Comment (ahmadrns) [inline easel/test/widget/file_type_icon_test.dart:16]: Mock `CreatorHubViewModel` itself

File: easel/test/widget/file_type_icon_test.dart [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+                  nft: MOCK_NFT,
+                );
+              },
+            ),
+          ),
+        );
+
+        final fileTypeIcon = find.byKey(const Key(kNFTTypeImageIconKey));
+        expect(fileTypeIcon, findsOneWidget);
+        await widgetTester.pump();
```
Comment (ahmadrns) [inline easel/test/widget/file_type_icon_test.dart:37]: `await widgetTester.pump()` should be before expect statement, there is no need to write it down after `expect`

File: easel/test/widget/file_type_icon_test.dart [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+                  nft: MOCK_PRICED_Video_NFT,
+                );
+              },
+            ),
+          ),
+        );
+
+        final fileTypeIcon = find.byKey(const Key(kNFTTypeVideoIconKey));
+        expect(fileTypeIcon, findsOneWidget);
+        await widgetTester.pump();
```
Comment (ahmadrns) [inline easel/test/widget/file_type_icon_test.dart:59]: Same here

File: easel/test/widget/file_type_icon_test.dart [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+                return NftGridViewItem(
+                  nft: MOCK_PRICED_AUDIO_NFT,
+                );
+              },
+            ),
+          ),
+        );
+
+        final fileTypeIcon = find.byKey(const Key(kNFTTypeAudioIconKey));
+        expect(fileTypeIcon, findsOneWidget);
```
Comment (ahmadrns) [inline easel/test/widget/file_type_icon_test.dart:80]: Same here

File: easel/test/widget/file_type_icon_test.dart [hunk 11]
```diff
@@ -0,0 +1,126 @@
...
+                return NftGridViewItem(
+                  nft: MOCK_PRICED_PDF_NFT,
+                );
+              },
+            ),
+          ),
+        );
+
+        final fileTypeIcon = find.byKey(const Key(kNFTTypePdfIconKey));
+        expect(fileTypeIcon, findsOneWidget);
```
Comment (ahmadrns) [inline easel/test/widget/file_type_icon_test.dart:101]: Same here

File: easel/test/widget/file_type_icon_test.dart [hunk 12]
```diff
@@ -0,0 +1,126 @@
...
+                return NftGridViewItem(
+                  nft: MOCK_PRICED_3D_NFT,
+                );
+              },
+            ),
+          ),
+        );
+
+        final fileTypeIcon = find.byKey(const Key(kNFTType3dModelIconKey));
+        expect(fileTypeIcon, findsOneWidget);
```
Comment (ahmadrns) [inline easel/test/widget/file_type_icon_test.dart:122]: Same here

File: easel/lib/screens/creator_hub/widgets/draft_list_tile.dart (hunk 4 above)
Comment (Kamran-Khan-Gene) [reply easel/lib/screens/creator_hub/widgets/draft_list_tile.dart:247]: @ahmadrns it is done using dart formatter. it is exactly 2 spaces.

File: easel/test/widget/gridview_banner_for_draft_published_test.dart [hunk 13]
```diff
@@ -28,8 +29,8 @@ void main() {
           await tester.setScreenSize();
           await tester.testAppForWidgetTesting(
             Scaffold(
-              body: ChangeNotifierProvider(
-                create: (ctx) => GetIt.I.get<CreatorHubViewModel>(),
+              body: ChangeNotifierProvider.value(
+                value: GetIt.I.get<CreatorHubViewModel>(),
```
Comment (ahmadrns) [inline easel/test/widget/gridview_banner_for_draft_published_test.dart:33]: There is no need to mock the `Repository` instead mock `CreatorHubViewModel` itself

File: easel/lib/screens/creator_hub/widgets/nfts_list_tile.dart (hunk 5 above)
Comment (Kamran-Khan-Gene) [reply easel/lib/screens/creator_hub/widgets/nfts_list_tile.dart:222]: same as i mentioned above

**Review by Amean — APPROVED** [review]

**Review by kjawadDeveloper2 — CHANGES_REQUESTED** [review]
I cannot run and test it because of being in another repository.  Removing my review.

**Pull request closed without merging**